build.sbt

```diff
@@ -31,20 +31,24 @@ val javacSettings = Seq(
   "-Xlint:unchecked"
 )
 
-def scalacOpts: Seq[String] =
-  Seq(
-    "-target:jvm-1.8",
+val scalacOpts = Def.setting[Seq[String]] {
+  val sv = scalaBinaryVersion.value
+
+  val common = Seq(
     "-deprecation",
     "-encoding",
     "UTF-8",
     "-feature",
-    "-unchecked",
-    "-Ywarn-unused:imports",
-    "-Xlint:nullary-unit",
-    "-Xlint",
-    "-Ywarn-dead-code",
+    "-unchecked"
   )
 
+  if (sv == "3") {
+    common
```

Comment:
@xuwei-k @cchantep Can we please add `-Xtarget:8` for Scala 3 here as well? It's the current default in Scala 3 anyway, but I think it's a good idea to make it explicit, so when we drop Java 8 we can change it to 11.
Also as soon as we drop Java 8 (meaning when we stop building with Java 8 as well) we should switch to `-release` anyway...

```sh
scalac -X
-Xtarget                    Emit bytecode for the specified version of the Java platform. This might produce bytecode that will break at runtime. When on JDK 9+,
                            consider -release as a safer alternative.
                            Choices: 8, 9, 10, 11, 12, 13, 14, 15, 16, 17.
```
See
https://github.com/lampepfl/dotty/issues/8634
https://github.com/lampepfl/dotty/pull/10746
config was rename just a week ago: 
was renamed a week ago:
https://github.com/lampepfl/dotty/pull/14606